src/js/background/context-menu.js

```diff
@@ -0,0 +1,371 @@
+    title: browser.i18n.getMessage("pageInputIconUseExistingAliasFromTheSite"),
+  }, 
+  useExistingAlias: {
+    id: "fx-private-relay-use-existing-aliases",
+    title: browser.i18n.getMessage("pageInputIconRecentAliases"),
+  }
+}
+
+const relayContextMenus = {
+  init: async (currentWebsite=null) => {
+    
+    if (!browser.menus) {
+      throw new Error(`Cannot create browser menus`);
+    }
+
+    const userApiToken = await browser.storage.local.get("apiToken");
+    const apiKeyInStorage = userApiToken.hasOwnProperty("apiToken");
+
+    if (!apiKeyInStorage) {
+      // User is not logged in. Do not do anything.
+      return;
+    }
+
+    // Reset any previously created menus
+    await browser.menus.removeAll();
+
+    // Generate aliases menu item
+    // If a user is maxed out/not premium, the generate item will be disabled.
+    const canUserGenerateAliases = await relayContextMenus.utils.getUserStatus.canGenerateMoreAliases();
+    canUserGenerateAliases ? relayContextMenus.menus.create(staticMenuData.generateAliasEnabled) : relayContextMenus.menus.create(staticMenuData.generateAliasDisabled);
+
+    const userHasSomeAliasesCreated = (await relayContextMenus.utils.getUserStatus.getNumberOfAliases() > 0);
+    
+    const aliases = await relayContextMenus.utils.getAliases();
+
+    // Create Use Existing Alias submenu
+    if (currentWebsite &&  await relayContextMenus.utils.getGeneratedForHistory(currentWebsite) && userHasSomeAliasesCreated ) {
+      await relayContextMenus.menus.create(staticMenuData.existingAlias, {
+        createExistingAliases: true,
+        parentMenu: staticMenuData.useExistingAliasFromWebsite,
+        exisitingSite: true,
+        currentWebsite
+      }, aliases);
+    } 
+
+    // Create "Recent Aliases…" menu
+    if ( userHasSomeAliasesCreated ) {
+      await relayContextMenus.menus.create(staticMenuData.existingAlias, {
+        createExistingAliases: true,
+        parentMenu: staticMenuData.useExistingAlias,
+        exisitingSite: false,
+        currentWebsite
+      }, aliases)
+    }
+
+    // Create "Manage all aliases" link
+    await relayContextMenus.menus.create(staticMenuData.manageAliases);
+
+    // Generate upgrade menu item for non-premium users
+    const canUserUpgradeToPremium = await relayContextMenus.utils.getUserStatus.canUpgradeToPremium();
+    if (canUserUpgradeToPremium) {
+      await relayContextMenus.menus.create([staticMenuData.upgradeToPremiumSeperator, staticMenuData.upgradeToPremium]);
+    }
+
+    // Set listerners
+    browser.storage.onChanged.addListener(relayContextMenus.listeners.onLocalStorageChange);        
+
+    // Refresh menus
+    await browser.menus.refresh();
+
+    return Promise.resolve(1)
+
+  },
+  listeners: {
+    onFillInAddressWithAliasId: async (info, tab) => {
+      // Trim the context menu id to get the alias reference ID
+      const selectedAliasId = info.menuItemId.replace(
+        "fx-private-relay-use-existing-alias_",
```

Comment:
Feels like a magic string. Can we move it into a constant value to help keep us from accidentally changing it in only 1 place in the future?

Reply:
31afe95